common/service/ipmi/include/ipmi.h

```diff
@@ -264,6 +264,7 @@ enum {
 	CMD_OEM_1S_GET_PCIE_CARD_STATUS = 0x76,
 	CMD_OEM_1S_GET_PCIE_CARD_SENSOR_READING = 0x77,
 	CMD_OEM_1S_SET_DEVICE_ACTIVE = 0x78,
+	CMD_OEM_1S_READ_DEVICE_FRU_DATA = 0x79,
```

Comment:
@KaidanWu-wiwynn Can you explain why we cannot use CMD_STORAGE_READ_FRUID_DATA here?
It has worked for GPv2 and GPv3 to read "Device FRUID".

Reply:
Hi @amithash , 

Thanks for your comments.
The CMD_STORAGE_READ_FRUID_DATA reads and writes fru data based on the fru id, but the method for BMC to read freya fru must include fru id and device id to determine which device to read fru data.

We have fixed the conflict and do the force update.
Please help to review again, thank you.

Regards,
Lora.

Reply:
The way I looked at the code on the BMC, https://github.com/facebook/openbmc/blob/helium/meta-facebook/meta-fby2/recipes-fby2/gpiointrd/files/gpiointrd.c#L293 its a bit confusing, they are using fru_id as a map to the device ID. If fru_id == 0, then its the main board's (or slot's) FRUID and if it is non-0, then it is mapped to the device ID.

Reply:
If you flatten the structure here (instead of fru_id + dev_id and use one dev_id within the board, which maps to a particular device, then you can reuse the old IPMI command instead of introducing yet another.

Reply:
Hi @amithash ,

Thanks for your comments.

There are two ways to get freya fru information：
1.  Take freya as a device ( Current implementation )
Use the new ipmi oem command CMD_OEM_1S_READ_DEVICE_FRU_DATA to get freya fru information according to fru id and device id.

2. Take freya as a fru
ACB fru id defination on BMC code：
https://github.com/facebookexternal/openbmc.wiwynn/blob/main/meta-facebook/meta-grandteton/recipes-grandteton/plat-libs/files/pal/pal_def.h#L272-L283

There are two freya on one ACCL card, so if we want to take freya as fru, we need to add 24 fru id.
And we can use ipmi command CMD_STORAGE_READ_FRUID_DATA to read fru information.

May I ask which way do you prefer?

Regards,
Lora.

Reply:
What I am saying is use the fru_id field of the CMD_STORAGE_READ_FRUID_DATA as equivalent of device ID.
When the BIC receives this command, you can make a determination of:
```
fru_id | what
0  | Main board FRU
1 | FREYA1
2 | FREYA2
3 | FREYA1:DEV0
4 | FREYA1:DEV1
....
```
From that perspective, we are essentially using the fru_id field of the command as a device ID.

Reply:
I am not saying we need to map the fru_id used in the CMD_STORAGE_READ_FRUID_DATA command to the actual FRU IDs used by the BMC. Probably the better name for the fru_id field of that command should be "component_id".

Reply:
Hi @amithash , 
OK. We will revise this feature based on you comments. Thank you.